onnx/version_converter/adapters/axis_input_to_attribute.h

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) ONNX Project Contributors
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "onnx/version_converter/adapters/adapter.h"
+
+namespace ONNX_NAMESPACE {
+namespace version_conversion {
+
+class AxisInputToAttribute : public Adapter {
+ public:
+  // Convert axis from input to attribute.
+  // axis_index: index of the axis input
+  // default_axis: default value of axis
+  explicit AxisInputToAttribute(
+      const std::string& op_name,
+      const OpSetID& initial,
+      const OpSetID& target,
+      int64_t axis_index,
+      int64_t default_axis)
+      : Adapter(op_name, initial, target) {
+    this->axis_index = axis_index;
+    this->default_axis = default_axis;
+      }
+
+  Node* adapt(std::shared_ptr<Graph> graph, Node* node) const override {
+    // Identify if axis is statically determined; if so, feed as attribute
+    const ArrayRef<Value*>& inputs = node->inputs();
+    // Get axis from initializer or constant operator
+    // Identify whether we have a Constant Op or an Initializer
+    // TODO(justinchuby): Avoid segfault
+    Value* const_val = inputs[this->axis_index];
+    // TODO(justinchuby): How do I check if axis is empty? In which cases it should take default
```

Comment:
@gramalingam I am working on the version converter. I was wondering how I can check if axis is an empty input?

Reply:
I think by checking for two conditions:
* That the `inputs.size() >= this->axis_index`, and
* checking that the `node_ptr->kind() != kUndefined`

Reply:
Moved to https://github.com/onnx/onnx/pull/5613 (WIP). Thanks!